hazelcast/src/test/java/com/hazelcast/spi/discovery/impl/DiscoveryServicePropertiesUtilTest.java

```diff
@@ -120,6 +121,25 @@ public void unknownProperty() {
         // throw exception
     }
 
+    @Test
+    public void nullOptionalProperty() {
```

Comment:
Minor: shouldn't you name the method "nullProperty"? I think it should work the same no matter if it's optional or not. If we decide to accept the `null` values, then in both cases they should be mapped.